### ISSUE TITLE
tensorflow keras random seeds not working

### ISSUE THREAD

**Issue by volvador** [tensorflow/tensorflow#22986]
Hello,
I am using Tensorflow 1.5 and its embedded Keras on CPU.
The following simple program does not produce the same results at every run despite setting the random seeds for keras and TF

    import numpy as np
    np.random.seed(123)

    import tensorflow as tf
    tf.set_random_seed(123)
    from tensorflow.python.keras.models import Sequential
    from tensorflow.python.keras.layers import Dense

    def create_model(input_dim, output_dim, training=True):
        from tensorflow.python.keras.models import Sequential
        from tensorflow.python.keras.layers import Dense
        hidden_dim = 8 
        model = Sequential()
        model.add(Dense(input_dim=input_dim, units=hidden_dim))
        model.add(Dense(units=output_dim, activation='linear'))
        model.compile(loss='mse', optimizer='adam')
        return model

    N = 100000
    X = np.random.normal(loc=0.0, scale=1.0, size=(N, 20))
    Y = np.random.normal(loc=0.0, scale=1.0, size=(N, 2)) 
    input_dim = X.shape[1]
    output_dim = Y.shape[1]

    tf.reset_default_graph()
    tf.set_random_seed(123)
    model = create_model(input_dim, output_dim)
    model.fit(X, Y, batch_size=100, epochs=1, verbose=1)

    Y_pred = model.predict(X, batch_size=1)

    print Y_pred[0:5]

**Comment by MartinKondor** [comment]
Try with plain random seed first at the top of the file with 1.
```
from random import seed
seed(1)
```
Btw, what is your python version?

**Comment by MartinKondor** [comment]
What if you use random as a class?
```
from random import Random

random = Random(123)
print random.random() 
```
This is working perfectly every time for me, with the same random values.

**Comment by volvador** [comment]
I think the problem is with the initialisation of random values inside Keras and Tensorflow. Please take the code I put, you will see that you do not reproduce same values

**Comment by harshini-gadige** [comment]
@volvador Hi, can you try with tf-nightly and see if you are running into the same problem.

**Comment by harshini-gadige** [comment]
@volvador  Is this still an issue ?  tf-nightly didn't work ?

**Comment by harshini-gadige** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
